### PR TITLE
Fix --use-current-runtime on Alpine

### PR DESCRIPTION
**Pull request by am11**
See https://github.com/dotnet/sdk/pull/22373#discussion_r985007320.

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets [hunk 1]
```diff
@@ -72,7 +72,8 @@ Copyright (c) .NET Foundation. All rights reserved.
                    '$(PublishAot)' == 'true'
                    )
                  )">
-    <RuntimeIdentifier>$(NETCoreSdkPortableRuntimeIdentifier)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="$(NETCoreSdkRuntimeIdentifier.StartsWith('linux-musl'))">$(NETCoreSdkRuntimeIdentifier)</RuntimeIdentifier>
+    <RuntimeIdentifier Condition="'$(RuntimeIdentifier)' == ''">$(NETCoreSdkPortableRuntimeIdentifier)</RuntimeIdentifier>
```
Comment (tmds) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:76]: We should instead fix `NETCoreSdkPortableRuntimeIdentifier` so it is `linux-musl-{arch}` on Alpine.

Comment (am11) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:76]: It's defined in the installer repo, correct?

Comment (tmds) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:76]: Yes. And it looks like the `x64` build gets it right:
```
    <NETCoreAppMaximumVersion>6.0</NETCoreAppMaximumVersion>
    <BundledNETCoreAppTargetFrameworkVersion>6.0</BundledNETCoreAppTargetFrameworkVersion>
    <BundledNETCoreAppPackageVersion>6.0.9</BundledNETCoreAppPackageVersion>
    <BundledNETStandardTargetFrameworkVersion>2.1</BundledNETStandardTargetFrameworkVersion>
    <BundledNETStandardPackageVersion>2.1.0</BundledNETStandardPackageVersion>
    <BundledNETCorePlatformsPackageVersion>6.0.5</BundledNETCorePlatformsPackageVersion>
    <BundledRuntimeIdentifierGraphFile>$(MSBuildThisFileDirectory)RuntimeIdentifierGraph.json</BundledRuntimeIdentifierGraphFile>
    <NETCoreSdkVersion>6.0.401</NETCoreSdkVersion>
    <NETCoreSdkRuntimeIdentifier>linux-musl-x64</NETCoreSdkRuntimeIdentifier>
    <NETCoreSdkPortableRuntimeIdentifier>linux-musl-x64</NETCoreSdkPortableRuntimeIdentifier>
```

Comment (tmds) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:76]: Unlike the `arm64` build:
```
    <NETCoreAppMaximumVersion>6.0</NETCoreAppMaximumVersion>
    <BundledNETCoreAppTargetFrameworkVersion>6.0</BundledNETCoreAppTargetFrameworkVersion>
    <BundledNETCoreAppPackageVersion>6.0.9</BundledNETCoreAppPackageVersion>
    <BundledNETStandardTargetFrameworkVersion>2.1</BundledNETStandardTargetFrameworkVersion>
    <BundledNETStandardPackageVersion>2.1.0</BundledNETStandardPackageVersion>
    <BundledNETCorePlatformsPackageVersion>6.0.5</BundledNETCorePlatformsPackageVersion>
    <BundledRuntimeIdentifierGraphFile>$(MSBuildThisFileDirectory)RuntimeIdentifierGraph.json</BundledRuntimeIdentifierGraphFile>
    <NETCoreSdkVersion>6.0.401</NETCoreSdkVersion>
    <NETCoreSdkRuntimeIdentifier>linux-musl-arm64</NETCoreSdkRuntimeIdentifier>
    <NETCoreSdkPortableRuntimeIdentifier>linux-arm64</NETCoreSdkPortableRuntimeIdentifier>
```

Comment (am11) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.RuntimeIdentifierInference.targets:76]: Yup, I found the same behavior. Do you know of quick way to test this installer patch:

```diff
--- a/src/redist/targets/GetRuntimeInformation.targets
+++ b/src/redist/targets/GetRuntimeInformation.targets
@@ -26,7 +26,7 @@
     <PropertyGroup>
       <ProductMonikerRid Condition=" '$(Rid)' == 'ubuntu.16.04-x64' OR
                                    '$(Rid)' == 'rhel.6-x64' OR
-                                   '$(Rid)' == 'linux-musl-x64' ">$(Rid)</ProductMonikerRid>
+                                   $(Rid.StartsWith('linux-musl') ">$(Rid)</ProductMonikerRid>
       <ProductMonikerRid Condition=" '$(ProductMonikerRid)' == '' ">$(OSName)-$(Architecture)</ProductMonikerRid>
 
       <PortableProductMonikerRid Condition=" '$(PortableProductMonikerRid)' == '' ">$(HostOSName)-$(Architecture)</PortableProductMonikerRid>
```

**Comment by am11** [comment]
Closing in favor of https://github.com/dotnet/installer/pull/14637.

**Pull request closed without merging**